src/app/utils/functions/deep-equal-objects/deep-equal-objects.ts

```diff
@@ -12,15 +12,22 @@ export const deepEqualObjects = (defaultObject, objectToCompare): boolean => {
   return defaultKeys.every(key => {
     const val1 = defaultObject[key];
     const val2 = objectToCompare[key];
+
     const areObjects = isObject(val1) && isObject(val2);
+    const areDates = isDate(val1) && isDate(val2);
 
-    const equalObjects = areObjects && deepEqualObjects(val1, val2);
+    const equalDates = areDates && val1?.toString() === val2?.toString();
+    const equalObjects = areObjects && !areDates && deepEqualObjects(val1, val2);
     const equalValues = !areObjects && val1 === val2;
 
-    return (equalObjects || equalValues);
+    return (equalObjects || equalValues || equalDates);
   });
 };
 
 const isObject = (object) => {
   return object != null && typeof object === 'object';
 };
+
+const isDate = (object) => {
+  return object instanceof Date; //&& Object.toString.call(object) === '[object Date]'
```

Comment:
Haha, I did a quick google to see how to best check `isDate` and found this SO answer which it looks like you found too: https://stackoverflow.com/a/643827
Why did you decide to go with the `instanceof` approach?

Reply:
Ha, good catch. I was checking something and i disabled it temporarily. And it was failing, because i didn't add `prototype`, so it should be here `Object.prototype.toString.call(object) === '[object Date]'`
I think i should remove instanceof and go with prototype solution.